lib/ui/painting.dart

```diff
@@ -1737,6 +1751,19 @@ class Image {
     return _image.toByteData(format: format);
   }
 
+  /// The color space that used by the [Image]'s colors.
+  ColorSpace get colorSpace {
+    final int colorSpaceValue = _image.colorSpace;
+    switch (colorSpaceValue) {
+      case 0:
+        return ColorSpace.sRGB;
+      case 1:
+        return ColorSpace.extendedSRGB;
+      default:
+        throw StateError('Unrecognized color space: $colorSpaceValue');
```

Comment:
I think since this value can only come from the engine, it would only be a programmer error on our side if it surfaced, rather than a StateError which implies the user used this code incorrectly.

Perhaps just assert?

Reply:
Yea good call.  I moved it to an `Exception`.  An assert could get ripped out, which we don't want here.

Reply:
Actually I think this should be an Error based on Bob's distinction: 
```
1. Exception: stuff you catch: runtime errors.
2. Error: stuff you don't catch: system and programmatic errors.
```
source: https://groups.google.com/a/dartlang.org/g/misc/c/lx9CXiV3o30/m/s5l_PwpHUGAJ

This isn't something you should catch, but isn't a user error either.  Error seems more apropos.

Reply:
StateError specifically though: "Should be used when this particular object is currently in a state which doesn't support the requested operation, but other similar objects might, or the object itself can later change its state to one which supports the operation."

Which doesn't really make sense to me because if this happens its almost certainly because the engine itself has a bug. Which is why perhaps we should assert and then return a default?

Reply:
There isn't a default value though.  We'd have to add a `ImageByteFormat.unknown` then make everyone handle that even though it's presence would be indicative of an error.  The documentation for `StateError` points to `UnsupportedError` "This [Error](https://api.flutter.dev/flutter/dart-core/Error-class.html) is thrown when an instance cannot implement one of the methods in its signature."  I think that sounds right.

Reply:
but again, this is _only_ going to happen if we make a mistake in the engine implementation. End users shouldn't even be aware of it which is why I think it should assert.

The closest error type would be something like InternalError but thats just a java thing.

Reply:
Right, asserts are another type of Error though, they just throw an `AssertionError`.  The downside is that they are often removed though so we'd need the `ImageByteFormat.unknown` which essentially makes end-users have to *catch* the engine error, which is what we don't want.  I'm happy to discuss it over a quick VC if you are still on the fence about `UnsupportedError`.

Reply:
I don't think it matters that much. Ideally no one will ever see this error so I don't think it matters

Reply:
Yep, hopefully not.  I switched it to `UnsupportedError` as in not having a skia image or impeller texture is not supported.  I think that makes the most sense.  Thanks.